app/server-routes/src/main/scala/org/bitcoins/server/routes/Server.scala

```diff
@@ -78,16 +91,67 @@ case class Server(
       }
     }
 
-  def start(): Future[Http.ServerBinding] = {
+  def start(): Future[ServerBindings] = {
     val httpFut =
       Http()
         .newServerAt(rpcbindOpt.getOrElse("localhost"), rpcport)
         .bindFlow(route)
     httpFut.foreach { http =>
       logger.info(s"Started Bitcoin-S HTTP server at ${http.localAddress}")
     }
-    httpFut
+    val wsFut = startWsServer()
+
+    for {
+      http <- httpFut
+      ws <- wsFut
+    } yield ServerBindings(http, ws)
+  }
+
+  private def startWsServer(): Future[Option[Http.ServerBinding]] = {
+    wsConfigOpt match {
+      case Some(wsConfig) =>
+        val httpFut =
+          Http()
+            .newServerAt(wsConfig.wsBind, wsConfig.wsPort)
+            .bindFlow(wsRoutes)
+        httpFut.foreach { http =>
+          logger.info(s"Started Bitcoin-S websocket at ${http.localAddress}")
+        }
+        httpFut.map(Some(_))
+      case None =>
+        Future.successful(None)
+    }
+  }
+
+  private val maxBufferSize: Int = 25
+
+  /** This will queue [[maxBufferSize]] elements in the queue. Once the buffer size is reached,
+    * we will drop the first element in the buffer
+    */
+  private val tuple = {
+    //from: https://github.com/akka/akka-http/issues/3039#issuecomment-610263181
+    //the BroadcastHub.sink is needed to avoid these errors
+    // 'Websocket handler failed with Processor actor'
+    Source
+      .queue[Message](maxBufferSize, OverflowStrategy.dropHead)
+      .toMat(BroadcastHub.sink)(Keep.both)
+      .run()
   }
+
+  def walletQueue: SourceQueueWithComplete[Message] = tuple._1
+  def source: Source[Message, NotUsed] = tuple._2
+
+  private def wsRoutes: Route = {
+    path("events") {
+      Directives.handleWebSocketMessages(wsHandler)
+    }
+  }
+
+  private def wsHandler: Flow[Message, Message, Any] = {
+    //we don't allow input, so use Sink.ignore
+    Flow.fromSinkAndSource(Sink.ignore, source)
```

Comment:
might be nice to error log and then ignore unexpected input, just to avoid silent failures